google-cloud-core/src/test/java/com/google/cloud/PolicyV3Test.java

```diff
@@ -0,0 +1,261 @@
+    assertEquals("etag", FULL_POLICY_V1.getEtag());
+    Policy policy = FULL_POLICY_V1.toBuilder().setBindings(BINDINGS_NO_CONDITIONS).build();
+    assertEquals(BINDINGS_NO_CONDITIONS, policy.getBindingsList());
+    assertEquals("etag", policy.getEtag());
+    assertEquals(1, policy.getVersion());
+
+    assertEquals(BINDINGS_WITH_CONDITIONS, FULL_POLICY_V3.getBindingsList());
+    assertEquals(3, FULL_POLICY_V3.getVersion());
+    assertEquals("etag", FULL_POLICY_V3.getEtag());
+    policy = FULL_POLICY_V3.toBuilder().setBindings(BINDINGS_WITH_CONDITIONS).build();
+    assertEquals(BINDINGS_WITH_CONDITIONS, policy.getBindingsList());
+    assertEquals("etag", policy.getEtag());
+    assertEquals(3, policy.getVersion());
+
+    assertEquals(BINDINGS_WITH_CONDITIONS, FULL_POLICY_V3_WITH_VERSION_1.getBindingsList());
+    assertEquals(1, FULL_POLICY_V3_WITH_VERSION_1.getVersion());
+    assertEquals("etag", FULL_POLICY_V3_WITH_VERSION_1.getEtag());
+    policy =
+        FULL_POLICY_V3_WITH_VERSION_1
+            .toBuilder()
+            .setBindings(BINDINGS_WITH_CONDITIONS)
+            .setVersion(3)
+            .build();
+    assertEquals(BINDINGS_WITH_CONDITIONS, policy.getBindingsList());
+    assertEquals("etag", policy.getEtag());
+    assertEquals(3, policy.getVersion());
+  }
+
+  @Test
+  public void removeMemberFromPolicy() {
+    assertEquals(3, FULL_POLICY_V3.getBindingsList().get(0).getMembers().size());
+    List<Binding> bindings = new ArrayList<>(FULL_POLICY_V3.getBindingsList());
+
+    for (int i = 0; i < bindings.size(); ++i) {
+      Binding binding = bindings.get(i);
+      if (binding.getRole().equals(VIEWER)) {
+        bindings.set(i, binding.toBuilder().removeMembers(ALL_USERS).build());
+        break;
+      }
+    }
+
+    Policy updatedPolicy = FULL_POLICY_V3.toBuilder().setBindings(bindings).build();
+    assertEquals(2, updatedPolicy.getBindingsList().get(0).getMembers().size());
+  }
+
+  @Test
+  public void addMemberFromPolicy() {
+    assertEquals(3, FULL_POLICY_V3.getBindingsList().get(0).getMembers().size());
+    List<Binding> bindings = new ArrayList<>(FULL_POLICY_V3.getBindingsList());
+
+    for (int i = 0; i < bindings.size(); ++i) {
+      Binding binding = bindings.get(i);
+      if (binding.getRole().equals(VIEWER)) {
+        bindings.set(i, binding.toBuilder().addMembers("user:example@example.com").build());
+      }
+    }
+
+    Policy updatedPolicy = FULL_POLICY_V3.toBuilder().setBindings(bindings).build();
+    assertEquals(4, updatedPolicy.getBindingsList().get(0).getMembers().size());
+  }
+
+  @Test
+  public void removeBindingFromPolicy() {
+    assertEquals(2, FULL_POLICY_V3.getBindingsList().size());
+    List<Binding> bindings = new ArrayList<>(FULL_POLICY_V3.getBindingsList());
+
+    for (int i = 0; i < bindings.size(); ++i) {
+      Binding binding = bindings.get(i);
+      if (binding.getRole().equals(EDITOR) && binding.getCondition() == null) {
+        bindings.remove(i);
```

Comment:
sorry for missing this in the previous review, let's switch this to iterator.remove as well